src/Authenticator/PrimaryKeySessionAuthenticator.php

```diff
@@ -0,0 +1,110 @@
+<?php
+declare(strict_types=1);
+
+namespace Authentication\Authenticator;
+
+use ArrayAccess;
+use Authentication\Identifier\IdentifierInterface;
+use Cake\Http\Exception\UnauthorizedException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Session Authenticator with only ID
+ */
+class PrimaryKeySessionAuthenticator extends SessionAuthenticator
+{
+    /**
+     * @param \Authentication\Identifier\IdentifierInterface $identifier
+     * @param array<string, mixed> $config
+     */
+    public function __construct(IdentifierInterface $identifier, array $config = [])
+    {
+        $config += [
+            'identifierKey' => 'key',
```

Comment:
If you're inheriting from `SessionAuthenticator` how does `identifierKey` interact with `fields`?

What is `key` here? I don't see that as a field in the `auth_users` table that your tests use. Is it primarily to be aligned with the identifier configuration?

Reply:
key here maps to the dataField of the identifier, this is how they pass in the data.

Reply:
fields is unused afaik.